projects/batfish/src/main/java/org/batfish/main/Batfish.java

```diff
@@ -3169,6 +3178,97 @@ private void populateFlowHistory(FlowHistory flowHistory, String environmentName
     }
   }
 
+  // this is a hack to deal with router names that are named ethX
+  private boolean normalInterfaceCompare(String if1, String if2) {
+    if (if1 == null || if2 == null) {
+      return true;
+    }
+
+    if (if2.startsWith("eth")) {
+      if2 = "Ethernet" + if2.substring(3);
+    }
+
+    return if1.equals(if2);
+  }
+
+  private void mergeConfigurations(Map<String, Configuration> configurations,
+      Map<String, Configuration> hostConfigurations) {
+    for (Entry<String, Configuration> e : hostConfigurations.entrySet()) {
+      Configuration hc = e.getValue();
+      Configuration c = configurations.get(e.getKey());
+      if (c == null) {
+        // no merge required
+        configurations.put(e.getKey(), hc);
+      } else {
+        // merge the host configuration with the router configuration
+
+        // mangle::PREROUTING is input
+        // mangle::POSTROUTING is output
+
+        // create an ACL for each iface, apply unspecified to all, and specific to each iface
+        IpAccessList prerouting = hc.getIpAccessLists().remove("mangle::PREROUTING");
+        IpAccessList postrouting = hc.getIpAccessLists().remove("mangle::POSTROUTING");
+
+        if (!hc.getIpAccessLists().isEmpty()) {
+          throw new BatfishException("Merging iptables rules for " + hc.getName()
+              + ": only mangle tables are supported");
+        }
+
+        if (prerouting != null) {
+          for (Interface i: c.getInterfaces().values()) {
+
+            String dbgName = c.getHostname() + ":" + i.getName();
+
+            List<IpAccessListLine> newRules = prerouting.getLines().stream()
```

Comment:
Since you should not add input/output interface to IpAccessListLine, you may be able to accomplish your intended purpose here by creating a host-config-specific IpAccessListLine type class like we have for the other vendors.